querybook/webapp/types.d.ts

```diff
@@ -21,6 +21,9 @@ declare global {
             mode?: 'replace';
             renderer: () => React.ReactElement;
         };
+        DATA_TABLE_SEARCH_CONFIG?: {
+            calculateInitialState: <T>(initialState: T) => typeof initialState;
```

Comment:
`<T>(initialState: T) => T`. However I think you want to restrict it to IDataTableSearchState